python/datasources/acs_poverty.py

```diff
@@ -311,80 +321,164 @@ def accumulate_acs_data(self, data):
     # Splits the in memory aggregation into dataframes
 
     def split_data_frames(self):
-        state_data = []
-        county_data = []
+        race_state_data = []
+        sex_state_data = []
+        age_state_data = []
+
+        race_county_data = []
+        sex_county_data = []
+        age_county_data = []
 
         # Extract keys from self.data Tuple
-        # (state_fip, County_fip, Age, Sex, Race): {PopulationObj}
+        # (state_fip, county_fip, Age, Sex, Race): {PopulationObj}
         for data, population in self.data.items():
             state_fip, county_fip, age, sex, race = data
 
             population = self.data[data]
-            above = population[PovertyPopulation.ABOVE]
             below = population[PovertyPopulation.BELOW]
 
+            if (age is not None) + (sex is not None) + (race is not None) != 1:
+                raise AssertionError(f"Invalid Tuple: {data}")
+
             if county_fip is None:
-                state_data.append(
-                    [
-                        state_fip,
-                        self.state_fips[state_fip],
-                        race,
-                        age,
-                        sex,
-                        above,
-                        below,
-                    ]
-                )
+                if race:
```

Comment:
In general, it's preferred to compare explicitly against None, e.g. if race is not None

Ditto for all of the comparisons like this below

Details: https://stackoverflow.com/questions/7816363/if-a-vs-if-a-is-not-none

Reply:
Okay I had that origionally but Josh asked me to remove it so I added it back